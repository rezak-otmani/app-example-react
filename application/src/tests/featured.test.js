import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Featured from '../views/featured/featured';


describe('Featured Component', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Featured />
    );
  });

           it('it should render Featured components ', () => {
              expect(wrapper.length).toBe(1);

          });

           it('it should render the title', () => {
             const item = wrapper.find('h1').first();
             expect(item.length).toEqual(1);
             expect(item.childAt(0).text()).toBe('USING DATO')
           });
            
           it('it should render images', () => {
             const item = wrapper.find('img');
             expect(item.length).toEqual(6);
           });

           const tree = renderer.create(
                      <Featured />
                          ).toJSON();
                  expect(tree).toMatchSnapshot();


});
