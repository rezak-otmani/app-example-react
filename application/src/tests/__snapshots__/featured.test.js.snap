// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="pusher"
>
  <div
    className="ui vertical masthead center aligned segment"
  >
    <div
      className="center aligned column"
    >
      <div
        className="ui sizer vertical segment"
      >
        <h1
          className="ui huge header"
        >
          USING DATO
        </h1>
      </div>
    </div>
    <div
      className="ui seven columns center aligned stackable grid"
    >
      <div
        className="column"
      >
        <img
          alt=""
          src="sat.png"
        />
      </div>
      <div
        className="column"
      >
        <img
          alt=""
          src="up.png"
        />
      </div>
      <div
        className="column"
      >
        <img
          alt=""
          src="tech2023.png"
        />
      </div>
      <div
        className="seven column row"
      >
        <div
          className="column"
        >
          <img
            alt=""
            src="le.png"
          />
        </div>
        <div
          className="column"
        >
          <img
            alt=""
            src="infodata.png"
          />
        </div>
        <div
          className="column"
        >
          <img
            alt=""
            src="info.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
